diff -Naur -x '*.orig' dhcp-4.2.2/client/dhclient.c dhcp-4.2.2-fix-format-errors/client/dhclient.c
@@ -153,9 +153,9 @@ static void
 usage(const char *sfmt, const char *sarg)
 {
 	log_info("%s %s", message, PACKAGE_VERSION);
-	log_info(copyright);
-	log_info(arr);
-	log_info(url);
+	log_info("%s", copyright);
+	log_info("%s", arr);
+	log_info("%s", url);
 
 	/* If desired print out the specific error message */
 #ifdef PRINT_SPECIFIC_CL_ERRORS
@@ -561,9 +561,9 @@ main(int argc, char **argv) {
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info(copyright);
-		log_info(arr);
-		log_info(url);
+		log_info("%s", copyright);
+		log_info("%s", arr);
+		log_info("%s", url);
 		log_info("%s", "");
 	} else {
 		log_perror = 0;
diff -Naur -x '*.orig' dhcp-4.2.2/relay/dhcrelay.c dhcp-4.2.2-fix-format-errors/relay/dhcrelay.c
--- dhcp-4.2.2/relay/dhcrelay.c	2011-05-10 15:07:37.000000000 +0200
+++ dhcp-4.2.2-fix-format-errors/relay/dhcrelay.c	2011-08-29 19:08:55.561183444 +0200
@@ -428,9 +428,9 @@
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info(copyright);
-		log_info(arr);
-		log_info(url);
+		log_info("%s", copyright);
+		log_info("%s", arr);
+		log_info("%s", url);
 	} else 
 		log_perror = 0;
 
diff -Naur -x '*.orig' dhcp-4.2.2/server/dhcpd.c dhcp-4.2.2-fix-format-errors/server/dhcpd.c
--- dhcp-4.2.2/server/dhcpd.c	2011-04-21 16:08:15.000000000 +0200
+++ dhcp-4.2.2-fix-format-errors/server/dhcpd.c	2011-08-29 19:08:55.563183419 +0200
@@ -398,9 +398,9 @@
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info (copyright);
-		log_info (arr);
-		log_info (url);
+		log_info ("%s", copyright);
+		log_info ("%s", arr);
+		log_info ("%s", url);
 	} else {
 		log_perror = 0;
 	}
@@ -1048,9 +1048,9 @@
 				tmp = log_perror;
 				log_perror = 0;
 				log_info("%s %s", message, PACKAGE_VERSION);
-				log_info(copyright);
-				log_info(arr);
-				log_info(url);
+				log_info("%s", copyright);
+				log_info("%s", arr);
+				log_info("%s", url);
 				log_perror = tmp;
 			} else
 				log_fatal("invalid log facility");
