Index: dhcp-4.4.1/configure
===================================================================
--- dhcp-4.4.1/configure
+++ dhcp-4.4.1/configure	2019-05-15 17:59:38.527555419 +0200
@@ -5724,7 +5724,7 @@
 ### the user configures an alternate value).  LOCALSTATEDIR is totally
 ### braindead.  No one uses /usr/local/var/db/ nor /usr/local/var/run, and
 ### they would be insane for suggesting it.  We need to look in /var/for
-### 'db' and 'state/dhcp' for db files, and /var/run for pid files by
+### 'db' and 'state/dhcp' for db files, and /run for pid files by
 ### default.
 ###
 if test "x$prefix" = xNONE; then
Index: dhcp-4.4.1/configure.ac
===================================================================
--- dhcp-4.4.1/configure.ac
+++ dhcp-4.4.1/configure.ac	2019-05-15 18:03:31.696021531 +0200
@@ -326,7 +326,7 @@
 ### the user configures an alternate value).  LOCALSTATEDIR is totally
 ### braindead.  No one uses /usr/local/var/db/ nor /usr/local/var/run, and
 ### they would be insane for suggesting it.  We need to look in /var/for
-### 'db' and 'state/dhcp' for db files, and /var/run for pid files by
+### 'db' and 'state/dhcp' for db files, and /run for pid files by
 ### default.
 ###
 AC_PREFIX_PROGRAM(dhcpd)
Index: dhcp-4.4.1/configure.ac+lt
===================================================================
--- dhcp-4.4.1/configure.ac+lt
+++ dhcp-4.4.1/configure.ac+lt	2019-05-15 18:05:05.288208627 +0200
@@ -327,7 +327,7 @@
 ### the user configures an alternate value).  LOCALSTATEDIR is totally
 ### braindead.  No one uses /usr/local/var/db/ nor /usr/local/var/run, and
 ### they would be insane for suggesting it.  We need to look in /var/for
-### 'db' and 'state/dhcp' for db files, and /var/run for pid files by
+### 'db' and 'state/dhcp' for db files, and /run for pid files by
 ### default.
 ###
 AC_PREFIX_PROGRAM(dhcpd)
Index: dhcp-4.4.1/includes/site.h
===================================================================
--- dhcp-4.4.1/includes/site.h
+++ dhcp-4.4.1/includes/site.h	2019-05-15 18:04:11.244100589 +0200
@@ -135,9 +135,9 @@
 
 /* Define this if you want the dhcpd.pid file to go somewhere other than
    the default (which varies from system to system, but is usually either
-   /etc or /var/run. */
+   /etc or /run. */
 
-/* #define _PATH_DHCPD_PID	"/var/run/dhcpd.pid" */
+/* #define _PATH_DHCPD_PID	"/run/dhcpd.pid" */
 
 /* Define this if you want the dhcpd.leases file (the dynamic lease database)
    to go somewhere other than the default location, which is normally
