diff -Naur -x '*.orig' dhcp-4.2.2/client/dhclient.c dhcp-4.2.2-fix-format-errors/client/dhclient.c
--- dhcp-4.2.2/client/dhclient.c	2011-07-01 13:58:53.000000000 +0200
+++ dhcp-4.2.2-fix-format-errors/client/dhclient.c	2011-08-29 19:08:55.560183457 +0200
@@ -420,9 +420,9 @@
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info(copyright);
-		log_info(arr);
-		log_info(url);
+		log_info("%s", copyright);
+		log_info("%s", arr);
+		log_info("%s", url);
 		log_info("%s", "");
 	} else {
 		log_perror = 0;
@@ -701,9 +701,9 @@
 static void usage()
 {
 	log_info("%s %s", message, PACKAGE_VERSION);
-	log_info(copyright);
-	log_info(arr);
-	log_info(url);
+	log_info("%s", copyright);
+	log_info("%s", arr);
+	log_info("%s", url);
 
 
 	log_fatal("Usage: dhclient "
diff -Naur -x '*.orig' dhcp-4.2.2/relay/dhcrelay.c dhcp-4.2.2-fix-format-errors/relay/dhcrelay.c
--- dhcp-4.2.2/relay/dhcrelay.c	2011-05-10 15:07:37.000000000 +0200
+++ dhcp-4.2.2-fix-format-errors/relay/dhcrelay.c	2011-08-29 19:08:55.561183444 +0200
@@ -428,9 +428,9 @@
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info(copyright);
-		log_info(arr);
-		log_info(url);
+		log_info("%s", copyright);
+		log_info("%s", arr);
+		log_info("%s", url);
 	} else 
 		log_perror = 0;

diff -Naur -x '*.orig' dhcp-4.2.2/server/dhcpd.c dhcp-4.2.2-fix-format-errors/server/dhcpd.c
--- dhcp-4.2.2/server/dhcpd.c	2011-04-21 16:08:15.000000000 +0200
+++ dhcp-4.2.2-fix-format-errors/server/dhcpd.c	2011-08-29 19:08:55.563183419 +0200
@@ -474,9 +474,9 @@
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info (copyright);
-		log_info (arr);
-		log_info (url);
+		log_info ("%s", copyright);
+		log_info ("%s", arr);
+		log_info ("%s", url);
 	} else {
 		quiet = 0;
 		log_perror = 0;
@@ -1096,9 +1096,9 @@
 					log_perror = 0;
 					log_info("%s %s",
 						 message, PACKAGE_VERSION);
-					log_info (copyright);
-					log_info (arr);
-					log_info (url);
+					log_info ("%s", copyright);
+					log_info ("%s", arr);
+					log_info ("%s", url);
 					log_perror = tmp;
 				}
 			} else
@@ -1205,8 +1205,8 @@
 static void
 usage(void) {
 	log_info("%s %s", message, PACKAGE_VERSION);
-	log_info(copyright);
-	log_info(arr);
+	log_info("%s", copyright);
+	log_info("%s", arr);
 
 	log_fatal("Usage: dhcpd [-p <UDP port #>] [-f] [-d] [-q] [-t|-T]\n"
 #ifdef DHCPv6
