diff -p -up dhcp-4.1.1/client/dhclient.c.not_a_string dhcp-4.1.1/client/dhclient.c
--- dhcp-4.1.1/client/dhclient.c.not_a_string	2010-01-07 19:47:40.000000000 -0200
+++ dhcp-4.1.1/client/dhclient.c	2010-02-25 11:30:19.000000000 -0300
@@ -381,9 +381,9 @@ main(int argc, char **argv) {
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info(copyright);
-		log_info(arr);
-		log_info(url);
+		log_info("%s", copyright);
+		log_info("%s", arr);
+		log_info("%s", url);
 		log_info("%s", "");
 	} else {
 		log_perror = 0;
@@ -647,9 +647,9 @@ main(int argc, char **argv) {
 static void usage()
 {
 	log_info("%s %s", message, PACKAGE_VERSION);
-	log_info(copyright);
-	log_info(arr);
-	log_info(url);
+	log_info("%s", copyright);
+	log_info("%s", arr);
+	log_info("%s", url);
 
 	log_error("Usage: dhclient %s %s",
 #ifdef DHCPv6
diff -p -up dhcp-4.1.1/relay/dhcrelay.c.not_a_string dhcp-4.1.1/relay/dhcrelay.c
--- dhcp-4.1.1/relay/dhcrelay.c.not_a_string	2010-01-07 19:48:02.000000000 -0200
+++ dhcp-4.1.1/relay/dhcrelay.c	2010-02-25 11:30:22.000000000 -0300
@@ -403,9 +403,9 @@ main(int argc, char **argv) {
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info(copyright);
-		log_info(arr);
-		log_info(url);
+		log_info("%s", copyright);
+		log_info("%s", arr);
+		log_info("%s", url);
 	} else {
 		quiet = 0;
 		log_perror = 0;
diff -p -up dhcp-4.1.1/server/dhcpd.c.not_a_string dhcp-4.1.1/server/dhcpd.c
--- dhcp-4.1.1/server/dhcpd.c.not_a_string	2010-02-25 11:29:56.000000000 -0300
+++ dhcp-4.1.1/server/dhcpd.c	2010-02-25 11:30:22.000000000 -0300
@@ -459,9 +459,9 @@ main(int argc, char **argv) {
 
 	if (!quiet) {
 		log_info("%s %s", message, PACKAGE_VERSION);
-		log_info (copyright);
-		log_info (arr);
-		log_info (url);
+		log_info ("%s", copyright);
+		log_info ("%s", arr);
+		log_info ("%s", url);
 	} else {
 		quiet = 0;
 		log_perror = 0;
@@ -1061,9 +1061,9 @@ void postconf_initialization (int quiet)
 					log_perror = 0;
 					log_info("%s %s",
 						 message, PACKAGE_VERSION);
-					log_info (copyright);
-					log_info (arr);
-					log_info (url);
+					log_info ("%s", copyright);
+					log_info ("%s", arr);
+					log_info ("%s", url);
 					log_perror = tmp;
 				}
 			} else
@@ -1170,8 +1170,8 @@ void postdb_startup (void)
 static void
 usage(void) {
 	log_info("%s %s", message, PACKAGE_VERSION);
-	log_info(copyright);
-	log_info(arr);
+	log_info("%s", copyright);
+	log_info("%s", arr);
 
 	log_fatal("Usage: dhcpd [-p <UDP port #>] [-f] [-d] [-q] [-t|-T]\n"
 #ifdef DHCPv6
